### PR TITLE
Remove dependency on the Miniion module.

### LINKED ISSUE #3: Remove the dependency on the deprecated Miniion module
As mentioned in the issue [here](https://github.com/MovingBlocks/Terasology/issues/2078), the [Miniion](https://github.com/Terasology/Miniion) module is now deprecated. MOO depends on Miniion for its `TaskManagementSystem` and the `MinionAI` system. The useful code related to the Task system in Miniion must be moved into MOO and the AI must be implemented to use the new behavior trees.

### PR DESCRIPTION
**Pull request by Naman-sopho**
Resolves #3 
MasterOfOreon used the Miniion module for `TaskManagementSystem` which assigned tasks to the Oreons and spawning Oreons from the menu. 
To do further, Spawning using `Portals`(Issue #4 ) and a new Task management system.
After merging this Oreon spawn from the Menu will not work, until #4 is sorted out.

### CODE REVIEW

File: src/main/java/org/terasology/managerInterface/nui/ManagerInterfaceHUDElement.java [hunk 1]
```diff
@@ -244,25 +221,7 @@ public void populateCreaturesMenu() {
         }
 
         EntityManager entityManager = CoreRegistry.get(EntityManager.class);
```
Comment (mkienenb) [inline src/main/java/org/terasology/managerInterface/nui/ManagerInterfaceHUDElement.java]: Note that all references to CoreRegistry.get(class) need to be replaced with either @In annotations (best idea) or context.get(class) codes (find-and-replace).  You'll probably need an @In annotation to pull in a context.

**Comment by mkienenb** [comment]
Still looks good to me, although you might want to verify that your @In values are actually non-null.

File: src/main/java/org/terasology/managerInterface/nui/ManagerInterfaceHUDElement.java [hunk 2]
```diff
@@ -43,6 +39,13 @@
 
     private static final Logger logger = LoggerFactory.getLogger(ManagerInterfaceHUDElement.class);
 
+    @In
+    private EntityManager entityManager;
+    @In
+    private ManagerInterfaceSystem managerInterfaceSystem;
+    @In
+    private PrefabManager prefMan;
```
Comment (vampcat) [inline src/main/java/org/terasology/managerInterface/nui/ManagerInterfaceHUDElement.java:47]: Change this to `prefabManager` please.
Unless it's a variable like `i` for a loop counter, I would recommend that you stick with descriptive names.

**Review by vampcat — CHANGES_REQUESTED** [review]
One minor request for renaming.

**Pull request closed without merging**